src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/2.diff.sql

```diff
@@ -95,3 +94,64 @@ AS
         
     COMMIT TRANSACTION
 GO
+
+/***************************************************************************************/
+-- STORED PROCEDURE
+--    DeleteCustomTag
+--
+-- DESCRIPTION
+--    Delete specific custom tag
+--
+-- PARAMETERS
+--     @tagPath
+--         * The custom tag path
+--     @dataType
+--         * the data type of custom tag. 0 -- String, 1 -- BigInt, 2 -- Double, 3 -- DateTime, 4 -- PersonName
+/***************************************************************************************/
+CREATE PROCEDURE dbo.DeleteCustomTag (
+    @tagPath VARCHAR(64),
+    @dataType TINYINT)
+AS
+
+    SET NOCOUNT     ON
+    SET XACT_ABORT  ON
+
+    BEGIN TRANSACTION
+        
+        -- Check if tag exsit
+        DECLARE @tagStatus TINYINT
+        DECLARE @tagKey TINYINT
+        SELECT @tagKey = TagKey, @tagStatus = TagStatus
+        FROM dbo.CustomTag WITH(HOLDLOCK) 
+        WHERE dbo.CustomTag.TagPath = @tagPath
+
+        IF @@ROWCOUNT = 0
+            THROW 50404, 'custom tag not found', 1 
+
+        -- check if status is Added
+        IF @tagStatus <> 1 
+            THROW 50412, 'custom tag is not in status Added', 1
+
+        -- Update status to deindexing
+        UPDATE dbo.CustomTag
+        SET TagStatus = 2
+        WHERE TagKey = @tagKey
+
+        -- Delete index data
+        IF @dataType = 0
+            DELETE FROM dbo.CustomTagString WHERE TagKey = @tagKey
+        ELSE IF @dataType = 1
+            DELETE FROM dbo.CustomTagBigInt WHERE TagKey = @tagKey
+        ELSE IF @dataType = 2
+            DELETE FROM dbo.CustomTagDouble WHERE TagKey = @tagKey
+        ELSE IF @dataType = 3
+            DELETE FROM dbo.CustomTagDateTime WHERE TagKey = @tagKey
+        ELSE
```

Comment:
Perhaps we should explicitly handle unknown data types by throwing an exception?

Reply:
I'm not sure whether we should handle it.
Adding exception could handle some unexpected situation, like we added a new datatype not the SQL is not updated yet. 
While in another side, this should not happen unless a bug. this is storage procedure is only used internally and we have the control on input to avoid such thing happen.

Reply:
In SQL perhaps this is fine to leave. If it is truly an impossible situation, I would add a `Debug.Assert` in C# to convey the precondition to other developers